text/trace/0220-messaging-semantic-conventions-span-structure.md

```diff
@@ -0,0 +1,656 @@
+spans can link to zero, one, or multiple producer spans.
+
+For single-message scenarios, and if the "Deliver" or "Receive" spans would be
+root spans of a new trace, the creation context may also be used as a parent on
+those operations in addition to being added as a link. Keeping single-messages
+operations in the same trace can greatly improve the user experience.
+
+#### Settlement
+
+Messages can be settled in a variety of different ways:
+
+* The intermediary settles the messages as it is sent to the consumer. No
+  settlement operations happen on the consumer.
+* The consumer settles a message without awaiting an acknowledgment from the
+  intermediary.
+* The consumer settles a message and awaits an acknowledgment from the
+  intermediary. This involves a round-trip exchange between the consumer and
+  the intermediary.
+
+Settlement operations on the consumer can either be triggered manually by the
+user, or can be triggered automatically by messaging SDKs based on return
+values of callbacks.
+
+A "Settle" span should be created for every settlement operation that happens
+on the consumer (at-least-once and exactly-once). SDKs will, in some cases,
+auto-settle messages in push-scenarios when messages are delivered via
+callbacks.
+
+"Settle" spans should link to creation context of the messages that are
+settled, when possible.
+
+No settlement span should be created for settlement scenarios that do not
+involve any settlement operations on the consumer side.
+
+## Proposed changes and additions to the messaging semantic conventions
+
+### Operation name
+
+The following operations related to messages are covered by these semantic
+conventions:
+
+| Operation name | Description |
+|----------------|-------------|
+| `publish`      | One or more messages are provided for publishing to an intermediary. |
+| `create`       | A message is created. |
+| `receive`      | One or more messages are requested by a consumer. |
+| `deliver`      | One or more messages are passed to a consumer. |
+| `settle`       | One or more messages are settled. |
+
+For further details about each of those operations refer to the [section about trace structure](#trace-structure).
+
+### Span kind
+
+[Span kinds](https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/trace/api.md#spankind)
+SHOULD be set according to the following table, based on the operation a span describes.
+
+| Operation name | Span kind|
+|----------------|-------------|
+| `publish`      | `PRODUCER`, if no `create` spans are present. |
+| `create`       | `PRODUCER` |
+| `receive`      | `CONSUMER` |
+| `deliver`      | `CONSUMER` |
+
+The kind of `settle` spans should be set according to the [generic specification about span kinds](https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/trace/api.md#spankind),
+e. g. it should be set to `CLIENT` if the `settle` spans models a synchronous call
+to the intermediary.
```

Comment:
Can we just recommend CLIENT? I don't see any other that makes sense in given the actual usages of span kinds in our other (non-messaging) semantic conventions, even if it would fit the abstract span kind definition maybe.